src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,306 @@
+
+            // Get repo info used for create/ update gitHub issue
+            Octokit.Repository repo = await client.Repository.Get(_options.Owner, _options.Repository);
+            try
+            {
+                using (ITransaction tx = _stateManager.CreateTransaction())
+                {
+                    if (await gitHubIssueEvaluator.ContainsKeyAsync(tx, (repositoryBranchUpdateHistory.Repository, repositoryBranchUpdateHistory.Branch)))
+                    {
+                        // issue exists so just update the issue
+                        var issueNumber = await gitHubIssueEvaluator.TryGetValueAsync(tx, (repositoryBranchUpdateHistory.Repository, repositoryBranchUpdateHistory.Branch));
+                        Logger.LogInformation("Updating a gitHub issue number : " + issueNumber + " for the error : " + repositoryBranchUpdateHistory.ErrorMessage +
+                            " for the repository : " + repositoryBranchUpdateHistory.Repository);
+                        if (issueNumber.HasValue)
+                        {
+                            try
+                            {
+                                var issue = await client.Issue.Get(_options.Owner, _options.Repository, issueNumber.Value);
+                                var issueUpdate = issue.ToUpdate();
+                                issueUpdate.Body = issue.Body + Environment.NewLine 
+                                    + Environment.NewLine + description.ToString();
+                                issueUpdate.State = ItemState.Open;
+                                await client.Issue.Update(_options.Owner ,_options.Repository, issueNumber.Value, issueUpdate);
+                            }
+                            catch(Exception ex)
+                            {
+                                Logger.LogError(ex.Message,"Unable to update issue number " + issueNumber + " in GitHub.");
+                            }
+                        }
+                        else
+                        {
+                            Logger.LogInformation("Unable to retrive the issuenumber from the reliable services.");
+                        }
+                    }
+                    else
+                    {
+                        string labelName = "DependencyUpdateError";
+                        Logger.LogInformation("Creating a gitHub issue for the error : " + repositoryBranchUpdateHistory.ErrorMessage +
+                            " for the repository : " + repositoryBranchUpdateHistory.Repository);
+                        var createIssue = new NewIssue("Dependency Update Error");
+                        createIssue.Body = description.ToString();
+                        createIssue.Labels.Add(labelName) ;
+                        try
+                        { 
+                            var issue = await client.Issue.Create(repo.Id, createIssue);
```

Comment:
Why does the other branch use the overload that take `(owner, repo`, but this one uses the `(repoId` method? It seems like the method that just passes two strings is preferable, because then we don't have to fetch all the repository information to make this call (this is the only place in this method the "repo" value is used)

Reply:
using repoId instead.